### ISSUE TITLE
CIM-XML representation of return value of Pull operations for empty result

### ISSUE THREAD

**Issue by andy-maier** [pywbem/pywbem#1320]
DSP0200 defines the return type of pull operations (Open* and Pull*) as a list of items (instances or instance paths).

Some of the pywbem testclient test cases return an empty IRETURNVALUE element for representing an empty list of items.

However, the following test cases omit the IRETURNVALUE element in that case:

* OpenEnumerateInstances4/5/6/7/8/9 in testsuite/testclient/openenumerateinstanceswithpath.yaml

IMO, these are invalid test cases and they should be fixed by adding an empty IRETURNVALUE element to the test response.

**Comment by andy-maier** [comment]
**Discussion point:** Do we agree that an empty IRETURNVALUE element is needed for representing an empty result list?

**Comment by KSchopmeyer** [comment]
ACTION: Karl - Determine how we implemented this in OpenPegasus.

For other operations we do not require IRETURNVALUE for the open/pull operations but for other operations we do.  Note that we implemented a flag in the parser to make this decision.

**Comment by andy-maier** [comment]
The current pywbem code for intrinsic operations behaves as follows:

* For Pull... and Open... operations, `_imethodcall()` is invoked with `response_params_rqd=True`, which causes no check for IRETURNVALUE to be made (see [cim_operations.py#L1684](https://github.com/pywbem/pywbem/blob/master/pywbem/cim_operations.py#L1684)). When processing the result, `_get_rslt_params()` tolerates both, an IRETURNVALUE element that wrappers the output parameters, or the output parameters directly (see [cim_operations.py#L2051](https://github.com/pywbem/pywbem/blob/master/pywbem/cim_operations.py#L2051)).
* For all other operations, `_imethodcall()` is invoked with the default `response_params_rqd=None` which causes the IRETURNVALUE element to be checked for presence (see [cim_operations.py#L1672](https://github.com/pywbem/pywbem/blob/master/pywbem/cim_operations.py#L1672)). Note that the comment about ERROR is incorrect.

Bottom line, the pywbem code has full support for IRETURNVALUE being optional in responses to Open... and Pull... operations.

**Comment by andy-maier** [comment]
**Discussion point:** Do we believe that pywbem needs to support IRETURNVALUE being optional in responses to Open... and Pull... operations.

Karl wanted to check whether OpenPegasus returns the IRETURNVALUE element in responses to Open... and Pull... operations.

**Comment by andy-maier** [comment]
Also, this issue exists only because DSP0200 is not clear about how a sequence of result items is to be represented, so a DMTF Mantis issue should be opened.

**Comment by andy-maier** [comment]
COnclusion from meeting today is that we abandon this issue, create a new one for the DMTF issue, and pursue the PR in context of issue #919.
